gluten-ut/spark34/src/test/scala/org/apache/spark/sql/GlutenDataFrameAggregateSuite.scala

```diff
@@ -279,4 +283,55 @@ class GlutenDataFrameAggregateSuite extends DataFrameAggregateSuite with GlutenS
       randn(Random.nextLong())
     ).foreach(assertNoExceptions)
   }
+
+  Seq(true, false).foreach {
+    value =>
+      testGluten(s"SPARK-31620: agg with subquery (whole-stage-codegen = $value)") {
```

Comment:
Do we need to also add this rewritten test for other spark versions? Generally, we do that even though duplicate code is introduced.

Reply:
> Do we need to also add this rewritten test for other spark versions? Generally, we do that even though duplicate code is introduced.

For this case I felt it's OK to just test with one version (as long as it's newer version, e.g., 3.4, 3.5). `collect_list` / `collect_set` in Spark shouldn't have actual differences among different Spark versions. Having it tested with all Spark version definitely provides better coverage but we can also make trade-off taking maintenance effort into account.